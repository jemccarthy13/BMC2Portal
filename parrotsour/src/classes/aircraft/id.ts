export enum IDMatrix {
  HOSTILE = "red",
  SUSPECT = "orange",
  ASSUME_FRIEND = "green",
  NEUTRAL = "purple",
  FRIEND = "blue",
}
