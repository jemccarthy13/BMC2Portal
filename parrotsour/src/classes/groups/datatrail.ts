import { Point } from "classes/point"

export enum SensorType {
    RADAR,
    IFF,
    RAW, // radar & IFF
    ARROW
}

export class DataTrail {
    private radarData: Point[] = []
    private iffData: Point[] = []

    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    constructor(startPos?: Point){
        // Initialize radar and iff based on start Pos
    }
}