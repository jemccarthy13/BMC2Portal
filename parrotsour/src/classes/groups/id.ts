export enum IDMatrix {
    HOSTILE = "red",
    SUSPECT = "#FA8072",
    ASSUME_FRIEND = "light green",
    NEUTRAL = "purple",
    FRIEND = "blue"
}