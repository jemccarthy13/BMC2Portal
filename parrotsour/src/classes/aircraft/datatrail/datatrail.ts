import { Point } from "../../point"
import { IDMatrix } from "../id"

// TODO -- DATATRAIL -- register? so that Aircraft can generate one of each type of datatrail?
export abstract class DataTrail {
  private startPos: Point

  constructor(startPos: Point) {
    this.startPos = startPos
  }

  public getStartPos(): Point {
    return this.startPos
  }

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  public move(_heading: number): void {
    //console.warn("Moving DataTrail in base class. Will have no effect.")
  }

  abstract getCenterOfMass(heading: number): Point

  abstract draw(
    ctx: CanvasRenderingContext2D,
    heading: number,
    id: IDMatrix
  ): void
}
