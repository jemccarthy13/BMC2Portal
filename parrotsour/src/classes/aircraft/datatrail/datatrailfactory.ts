import { Point } from "../../point"
import { ArrowDataTrail } from "./arrowdatatrail"
import { DataTrail } from "./datatrail"
import { RawDataTrail } from "./rawdatatrail"
import { SensorType } from "./sensortype"

class defaultDataTrail extends DataTrail {
  create(startPos: Point) {
    return new defaultDataTrail(startPos)
  }
  getCenterOfMass(): Point {
    throw new Error("Method not implemented.")
  }
  draw(): void {
    throw new Error("Method not implemented.")
  }
}

export class DataTrailFactory {
  private static trailMap = new Map<
    SensorType,
    (startPos: Point) => DataTrail
  >()

  // TODO - DATATRAIL -- self-registering DataTrail child classes, if possible
  // static function register(type, createFunction)
  // each child has create(startPos) => return new [subtype]DataTrail
  static create(type: SensorType, startPos: Point, heading: number): DataTrail {
    switch (type) {
      case SensorType.ARROW:
        return new ArrowDataTrail(startPos)
      case SensorType.RAW:
        return new RawDataTrail(startPos, heading)
    }
  }
}
