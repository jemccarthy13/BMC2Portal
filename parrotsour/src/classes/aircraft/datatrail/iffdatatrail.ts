import { Point } from "../../point"
import { IDMatrix } from "../id"
import { DataTrail } from "./datatrail"

export class IFFDataTrail extends DataTrail {
  private iffData: Point[] = []

  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  getCenterOfMass(_heading: number): Point {
    return this.iffData[this.iffData.length - 1]
  }

  draw(ctx: CanvasRenderingContext2D, heading: number, id: IDMatrix): void {
    if (id !== IDMatrix.SUSPECT && id !== IDMatrix.HOSTILE) {
      //console.log("draw iff")
    }
    // error but don't break the runtime
  }
}
