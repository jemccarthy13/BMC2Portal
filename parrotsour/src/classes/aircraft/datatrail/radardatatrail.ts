import { drawBullseye } from "../../../canvas/draw/drawutils"
import { PaintBrush } from "../../../canvas/draw/paintbrush"
import {
  headingToRadians,
  PIXELS_TO_NM,
  toRadians,
} from "../../../utils/psmath"
import { Point } from "../../point"
import { IDMatrix } from "../id"
import { DataTrail } from "./datatrail"

export class RadarDataTrail extends DataTrail {
  private radarPoints: Point[] = []

  private LEN_TRAIL: number = 7 * PIXELS_TO_NM
  constructor(startPos: Point, heading: number) {
    super(startPos)

    const starty = startPos.y
    const startx = startPos.x
    const vector = headingToRadians(heading)
    const endy = starty + this.LEN_TRAIL * -Math.sin(vector.radians)
    const endx = startx + this.LEN_TRAIL * Math.cos(vector.radians)

    const offsetX = (endx - startx) / PIXELS_TO_NM
    const offsetY = (endy - starty) / PIXELS_TO_NM

    let xPos = startx
    let yPos = starty
    const rdrPts: Point[] = []
    // draw the radar trail
    if (!this.radarPoints || this.radarPoints.length === 0) {
      for (let mult = 0; mult < 6; mult++) {
        // add a bit of jitter with randomness
        const jit = 3
        xPos = startx + offsetX * mult + jit * Math.random() + Math.random()
        yPos = starty + offsetY * mult + jit * Math.random() + Math.random()
        const rdrPt = new Point(xPos, yPos)
        rdrPts.push(rdrPt)
      }
      this.radarPoints = rdrPts
    }
  }

  getCenterOfMass(heading: number): Point {
    return this._getOnePlotAhead()
  }

  public move(heading: number): void {
    // this logic should be in move()
    // take the first point out of the radar trail
    const newRdrPts = this.radarPoints.slice(1)

    const startPos = this.getStartPos()
    const startx = startPos.x
    const starty = startPos.y
    const vector = headingToRadians(heading)

    const endy = starty + this.LEN_TRAIL * -Math.sin(vector.radians)
    const endx = startx + this.LEN_TRAIL * Math.cos(vector.radians)
    const offsetX = (endx - startx) / PIXELS_TO_NM
    const offsetY = (endy - starty) / PIXELS_TO_NM

    // add a bit of jitter with randomness
    const jit = 3

    const xPos = startx + offsetX * 6 + jit * Math.random() + Math.random()
    const yPos = starty + offsetY * 6 + jit * Math.random() + Math.random()
    newRdrPts.push(new Point(xPos, yPos))
    this.radarPoints = newRdrPts
  }

  draw(ctx: CanvasRenderingContext2D, heading: number, id: IDMatrix): void {
    // Draw radar dots
    ctx.strokeStyle = "#FF8C00"
    ctx.beginPath()
    this.radarPoints.forEach((pt) => {
      ctx.beginPath()
      ctx.moveTo(pt.x, pt.y)
      ctx.lineTo(pt.x - 3, pt.y - 3)
      ctx.stroke()
      ctx.stroke()
    })

    this._drawSymbology(ctx, id)
  }

  public getDataTrail(): Point[] {
    return this.radarPoints
  }

  _getOnePlotAhead(): Point {
    const cPt = this.radarPoints[this.radarPoints.length - 1]
    const pPt = this.radarPoints[this.radarPoints.length - 2]

    const deltX = cPt.x - pPt.x
    const deltY = cPt.y - pPt.y

    return new Point(cPt.x + deltX, cPt.y + deltY)
  }

  _drawSymbology(ctx: CanvasRenderingContext2D, id: IDMatrix): void {
    const plotAhead = this._getOnePlotAhead()

    ctx.strokeStyle = id

    if (id === IDMatrix.FRIEND) {
      // draw friend symbology (blue arc/upside-down U)
      ctx.moveTo(plotAhead.x - 2.5, plotAhead.y - 2.5)
      ctx.beginPath()
      ctx.arc(
        plotAhead.x - 2.5,
        plotAhead.y - 0.5,
        4,
        toRadians(170),
        toRadians(10)
      )
      ctx.stroke()
    } else {
      ctx.beginPath()
      const headX = plotAhead.x - 3
      const headY = plotAhead.y - 3

      const leftX = headX + 5 * Math.cos(toRadians(240))
      const leftY = headY + 5 - Math.sin(toRadians(240))
      ctx.moveTo(headX, headY)
      ctx.lineTo(leftX, leftY)

      const rightX = headX + 5 * Math.cos(toRadians(300))
      const rightY = headY + 5 - Math.sin(toRadians(300))
      ctx.moveTo(headX, headY)
      ctx.lineTo(rightX, rightY)
      ctx.stroke()
      ctx.stroke()
    }
  }
}
