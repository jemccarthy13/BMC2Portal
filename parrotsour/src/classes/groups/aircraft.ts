import { DataTrail, SensorType } from "./datatrail"
import { GroupParams } from "./group"
import { IDMatrix } from "./id"
import { AircraftIntent, IntentParams } from "./intent"
import { Point } from "../point"
import Tasking from "./tasking"

import { PaintBrush } from "../../canvas/draw/paintbrush"
import {
  getDegDeltaBetween,
  headingToRadians,
  randomNumber,
} from "../../utils/psmath"

/*
 * TODO -- For EWI fill-ins, implement more AC types?
 */
export enum ACType {
  FTR,
  RPA,
  TANK,
  AEW,
}

interface AircraftParams extends GroupParams {
  alt: number
  type: ACType
}

/**
 * TODO -- AIRCRAFT -- consider moving to a structure like
 * https://khalilstemmler.com/blogs/typescript/getters-and-setters/
 * To reduce the number of functions provided in the aircraft "api"
 */
export class Aircraft {
  // AASI (alt, aspect, speed, ID)
  private altitude: number
  private heading: number
  // TODO -- implement aircraft speed
  // private speed: number
  private id: IDMatrix

  // TODO -- implement radar/iff data
  // Radar and IFF data associated with this group
  private dataTrail: DataTrail

  private type: ACType

  private startPos = new Point(0, 0)

  private intent = new AircraftIntent()
  private tasking: Tasking | undefined

  private ctx: CanvasRenderingContext2D | undefined

  constructor(p?: Partial<AircraftParams>) {
    if (!p) p = {}
    // temporary
    this.dataTrail = new DataTrail()
    //if (p) { // why would we need to pass in a DataTrail if it's associated with this aircraft?
    // this.dataTrail = new DataTrail()
    //}

    this.heading = p.hdg || 90
    this.id = p.id || IDMatrix.HOSTILE
    this.type = p.type || ACType.FTR

    // Set current position
    this.startPos.x = p.sx || randomNumber(1, 100)
    this.startPos.y = p.sy || randomNumber(1, 100)

    let low = 15
    let hi = 45
    if (this.type === ACType.RPA) {
      low = 0o5
      hi = 18
    }

    // Set AACSI (alt, aspect, # contacts, speed, ID)
    this.altitude = p.alt || randomNumber(low, hi)

    this.ctx = p.ctx || undefined
  }

  public getAltitude(): number {
    return this.altitude
  }

  public getStartPos(): Point {
    return this.startPos
  }

  /**
   * Get the center of mass position for this group.
   *
   * The type of data trail dictates where the center of mass is.
   * (i.e. radar - last sensed radar vs. arrow - loc of arrow head)
   *
   * @param ctx Current drawing context
   * @param dataStyle (Radar & IFF) | Arrows - must explicitly be set if drawing
   *                  anything other than arrows
   */
  private _getCenterOfMass(
    ctx?: CanvasRenderingContext2D,
    dataStyle?: SensorType
  ): Point {
    dataStyle = dataStyle || SensorType.ARROW
    if (dataStyle === SensorType.ARROW) {
      return this._getCenterOfMassArrow(ctx)
    } else {
      // return this.snsrData.current
      throw "help"
      //return Point.DEFAULT
    }
  }

  /**
   * The "center of mass" for a singular aircraft is the end point
   * of the Arrow.
   */
  private _getCenterOfMassArrow(ctx?: CanvasRenderingContext2D): Point {
    const vector = headingToRadians(this.getHeading())
    let dist = 10

    const context = ctx || this.ctx
    if (context) {
      dist = Math.floor(context.canvas.width / (context.canvas.width / 20))
    }

    return new Point(
      Math.floor(this.startPos.x + 1.2 * dist * Math.cos(vector.radians)),
      Math.floor(this.startPos.y + 1.2 * dist * -Math.sin(vector.radians))
    )
  }

  getCenterOfMass(): Point {
    return this._getCenterOfMass()
  }

  public getHeading(): number {
    return this.heading
  }

  public setCurHeading(hdg: number): void {
    this.heading = hdg
  }

  getIDMatrix(): IDMatrix {
    return this.id
  }

  setIDMatrix(newID: IDMatrix): void {
    this.id = newID
  }

  getType(): ACType {
    return this.type
  }

  draw(context: CanvasRenderingContext2D, dataStyle: SensorType): void {
    const brush: PaintBrush = new PaintBrush(context)
    brush.drawAircraftSensorData(
      this,
      this.startPos.x,
      this.startPos.y,
      dataStyle
      //groups[x].radarPoints,
      //groups[x].iffPoints,
      //groups[x].drawnRadar
    )
  }

  /**
   * Move the arrow once based on the current heading / vector.
   */
  move(): void {
    // if (this.isCapping()) return
    this.turnToTarget()

    // convert heading to radians and calculate how much arrow needs to move
    const rads: number = headingToRadians(this.getHeading()).radians
    const offsetX: number = 7 * Math.cos(rads)
    const offsetY: number = -7 * Math.sin(rads)

    // apply offsets based on start/end position
    this.startPos.x += offsetX
    this.startPos.y += offsetY
  }

  turnToTarget(): void {
    if (!this.intent.atFinalDestination()) {
      const tgtPos = this.intent.getNextRoutingPoint()
      this.intent.setDesiredHeading(
        this.getCenterOfMass().getBR(tgtPos).bearingNum
      )
    }

    const turnDegrees = getDegDeltaBetween(
      this.getHeading(),
      this.intent.getDesiredHeading()
    )

    let divisor = 7
    const absDelt = Math.abs(turnDegrees)
    if (absDelt >= 90) {
      divisor = 15
    } else if (absDelt < 7) {
      divisor = 1
    }

    this.setCurHeading(this.getHeading() + turnDegrees / divisor)
  }

  updateIntent(newIntent: Partial<IntentParams>): void {
    this.intent.updateIntent(newIntent)
  }

  /**
   * Routing logic
   */

  doNextRouting(): void {
    if (!this.intent.atFinalDestination()) {
      if (this.intent.atNextRoutingPoint(this.getCenterOfMass())) {
        this.intent.removeRoutingPoint()
      }
      const nextPt = this.intent.getNextRoutingPoint()
      this.updateIntent({
        desiredHeading: this.getCenterOfMass().getBR(nextPt).bearingNum,
      })
    } else {
      throw "Need to make capping logic compatible with Intent & class structure."
    }
  }

  hasRouting(): boolean {
    return !this.intent.atFinalDestination()
  }

  getNextRoutingPoint(): Point {
    return this.intent.getNextRoutingPoint()
  }

  addRoutingPoint(pt: Point): void {
    this.intent.addRoutingPoint(pt)
  }

  /**
   * If not at desired altitude, increment to climb/descend towards intended altitude.
   */
  doNextAltChange(): void {
    const atDesiredAlt = this.intent.getDesiredAltitude() === this.getAltitude()
    if (!atDesiredAlt) {
      if (this.intent.getDesiredAltitude() > this.getAltitude()) {
        this.altitude += 0.5
      } else {
        this.altitude += -0.5
      }
    }
  }

  /**
   * Tasking logic section
   */

  isTasked(): boolean {
    return this.tasking !== undefined
  }

  clearTasking(): void {
    this.tasking = undefined
  }

  setTasking(task: Tasking): void {
    this.tasking = task
  }

  getTasking(): Tasking | undefined {
    return this.tasking
  }
}
