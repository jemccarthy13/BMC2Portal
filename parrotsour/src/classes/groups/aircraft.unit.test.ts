import { Point } from "../point"
import { ACType, Aircraft } from "./aircraft"
import { SensorType } from "./datatrail"
import { IDMatrix } from "./id"
import Tasking from "./tasking"
// eslint-disable-next-line @typescript-eslint/no-var-requires
const canvasSerializer = require("jest-canvas-snapshot-serializer")

expect.addSnapshotSerializer(canvasSerializer)
describe("Aircraft", () => {
  const canvas = document.createElement("canvas")
  canvas.width = 10
  canvas.height = 30
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const ctx: CanvasRenderingContext2D = canvas.getContext("2d")!
  ctx.fillStyle = "white"
  ctx.fillRect(0, 0, 0, 0)

  it("constructs_FTR_correctly", () => {
    const acft = new Aircraft()
    expect(acft.getHeading()).toEqual(90)
    expect(acft.getAltitude()).toBeLessThanOrEqual(45)
    expect(acft.getAltitude()).toBeGreaterThanOrEqual(15)
    expect(acft.getIDMatrix()).toEqual(IDMatrix.HOSTILE)
    expect(acft.getNextRoutingPoint()).toEqual(undefined)
    expect(acft.getType()).toEqual(ACType.FTR)
    const startPos = acft.getStartPos()
    expect(startPos.x).toBeLessThanOrEqual(100)
    expect(startPos.y).toBeLessThanOrEqual(100)
    expect(startPos.x).toBeGreaterThanOrEqual(1)
    expect(startPos.y).toBeGreaterThanOrEqual(1)
  })

  it("constructs_RPA_correctly", () => {
    const acft = new Aircraft({ type: ACType.RPA })
    expect(acft.getAltitude()).toBeLessThanOrEqual(18)
    expect(acft.getAltitude()).toBeGreaterThanOrEqual(0o5)
  })

  it("computes_center_of_mass", () => {
    const bluePos = new Point(50, 50)
    const acft = new Aircraft({ sx: bluePos.x, sy: bluePos.y, ctx: ctx })
    expect(acft.getCenterOfMass()).toEqual({ x: 74, y: 50 })

    acft.setCurHeading(180)
    expect(acft.getCenterOfMass()).toEqual({ x: 49, y: 74 })
  })

  it("draws_Hid_correctly", () => {
    const bluePos = new Point(5, 5)
    const acft = new Aircraft({
      sx: bluePos.x,
      sy: bluePos.y,
      ctx: ctx,
      hdg: 180,
    })
    acft.draw(ctx, SensorType.ARROW)
    expect(canvas).toMatchSnapshot()
  })

  it("draws_Sid_correctly", () => {
    const bluePos = new Point(5, 5)
    const acft = new Aircraft({
      sx: bluePos.x,
      sy: bluePos.y,
      ctx: ctx,
      hdg: 180,
    })
    acft.setIDMatrix(IDMatrix.SUSPECT)
    acft.draw(ctx, SensorType.ARROW)
    expect(canvas).toMatchSnapshot()
  })

  it("draws_Nid_correctly", () => {
    const bluePos = new Point(5, 5)
    const acft = new Aircraft({
      sx: bluePos.x,
      sy: bluePos.y,
      ctx: ctx,
      hdg: 180,
    })
    acft.setIDMatrix(IDMatrix.NEUTRAL)
    acft.draw(ctx, SensorType.ARROW)
    expect(canvas).toMatchSnapshot()
  })

  it("draws_AFid_correctly", () => {
    const bluePos = new Point(5, 5)
    const acft = new Aircraft({
      sx: bluePos.x,
      sy: bluePos.y,
      ctx: ctx,
      hdg: 180,
    })
    acft.setIDMatrix(IDMatrix.ASSUME_FRIEND)
    acft.draw(ctx, SensorType.ARROW)
    expect(canvas).toMatchSnapshot()
  })

  it("intent_alt_functions", () => {
    const bluePos = new Point(5, 5)
    const acft = new Aircraft({
      sx: bluePos.x,
      sy: bluePos.y,
      ctx: ctx,
      hdg: 180,
      alt: 30,
    })
    acft.updateIntent({
      desiredAlt: 10,
    })
    acft.doNextAltChange()
    expect(acft.getAltitude()).toEqual(29.5)
    acft.updateIntent({
      desiredAlt: 40,
    })
    acft.doNextAltChange()
    expect(acft.getAltitude()).toEqual(30)
    acft.updateIntent({
      desiredAlt: 30,
    })
    acft.doNextAltChange()
    expect(acft.getAltitude()).toEqual(30)
  })

  it("set_task_functions", () => {
    const acft = new Aircraft()

    const tasking = new Tasking({
      locationXY: new Point(0, 0),
      locationStr: "88LC",
      description: "BOMBING STUFF",
    })
    acft.setTasking(tasking)
    expect(acft.isTasked()).toEqual(true)
    expect(acft.getTasking()).toEqual(tasking)

    acft.clearTasking()
    expect(acft.isTasked()).toEqual(false)
    expect(acft.getTasking()).toEqual(undefined)
  })

  it("turns_to_target", () => {
    // the turn goes off of center of mass, so here we position the aircraft
    // such that the turn math becomes easier ('desired Hdg' = 90, so the updated
    // heading is easier to expect)
    const acft = new Aircraft({
      sx: 25,
      sy: 38,
      hdg: 180,
    })
    acft.addRoutingPoint(new Point(50, 50))
    acft.turnToTarget()
    expect(acft.getHeading()).toEqual(180 - 90 / 15)
    expect(acft.getStartPos()).toEqual(new Point(25, 38))
  })

  it("doesnt_turn_at_target", () => {
    const dest = new Point(50, 50)
    const acft = new Aircraft({
      sx: dest.x,
      sy: 38,
      hdg: 180,
    })
    acft.turnToTarget() // expect nothing to change
    expect(acft.getHeading()).toEqual(180)
    expect(acft.getStartPos()).toEqual(new Point(50, 38))
  })

  it("sharp_turn_near_desired_hdg", () => {
    const dest = new Point(50, 50)
    const acft = new Aircraft({
      sx: dest.x,
      sy: dest.y,
      hdg: 180,
    })
    acft.updateIntent({ desiredHeading: 175 })
    acft.turnToTarget() // < 7 turn delta is snap to desired (to avoid data trail sin wave)
    expect(acft.getHeading()).toEqual(175)
    expect(acft.getStartPos()).toEqual(dest)
  })

  it("moves_along_heading", () => {
    const dest = new Point(50, 50)
    const acft = new Aircraft({
      sx: dest.x,
      sy: 38,
      hdg: 180,
    })
    acft.move() // move once
    expect(acft.getHeading()).toEqual(180)
    expect(acft.getStartPos()).toEqual(new Point(50, 45))
  })

  it("turns_during_move_if_intent", () => {
    const dest = new Point(50, 50)
    const acft = new Aircraft({
      sx: dest.x,
      sy: 38,
      hdg: 180,
    })
    acft.updateIntent({ desiredHeading: 135 })
    acft.move() // move once
    expect(acft.getHeading()).toEqual(180 - (180 - 135) / 7)
    expect(acft.getStartPos()).toEqual(new Point(50, 45))
  })
})
