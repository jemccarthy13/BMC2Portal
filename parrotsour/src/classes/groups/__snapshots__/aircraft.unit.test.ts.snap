// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aircraft draws_AFid_correctly 1`] = `
<canvas
  data-snapshot-image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAeCAYAAAAVdY8wAAAABmJLR0QA/wD/AP+gvaeTAAAApUlEQVQ4jWNgGM5AsIjvn2AR3z90cSZiDRhVODwUVgcr7VwVqq2CTyELAwMDQ4fmQzfZ90K38NnAxMDAwBB1V/HVA5HXjK0BSlNxKWSEMTgrOP4z/2NiYP3D8p+BgYHhfd8nFNPhnKrbssu+cH1j+ML1nZEBC0ARlMoT+vdM8B2j4Ge+/zhNZGBgYGh6LmmBy40YwDFB4Se2PIMVBMUqfyLaZNoDANZrK4UBkSJ0AAAAAElFTkSuQmCC"
  height="30"
  width="10"
/>
`;

exports[`Aircraft draws_Hid_correctly 1`] = `
<canvas
  data-snapshot-image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAeCAYAAAAVdY8wAAAABmJLR0QA/wD/AP+gvaeTAAAAe0lEQVQ4je3TsQ3CUAxF0RM6KlZgByQKBmApWuZgH2oWyASRaEAIiUfzCxSR/DQ0Ea+y7KvrxmbOCW1o+/3FVMEfnAcY9mFZpcMjnEs9ej0HbMNmSNZ8WDs8cS+D9dD6XUi4fVvdhy8FroKr8Kq+QsMVp1Fbz3ycDP8+b4G2KENdAA1tAAAAAElFTkSuQmCC"
  height="30"
  width="10"
/>
`;

exports[`Aircraft draws_Nid_correctly 1`] = `
<canvas
  data-snapshot-image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAeCAYAAAAVdY8wAAAABmJLR0QA/wD/AP+gvaeTAAAAnklEQVQ4jWNgGM5gglDBvwlCBf/QxZmINWBU4fBQeNzUYt5jS0tpggr3nHdJ3P5I9xFBhRY6p68/eyrJdNLCpBqXQkYYo4Or/D8T818GNtY//xkYGBgK3k1AcT+c46x7oPvbZx6Gr194GBmwALhC05Mny8TEX///9YsZtxthwFXhjA4jVvPQFKqePHlNV+PqRxz+wQTb9L0fE62Y9gAA5OstCQQiAdYAAAAASUVORK5CYII="
  height="30"
  width="10"
/>
`;

exports[`Aircraft draws_Sid_correctly 1`] = `
<canvas
  data-snapshot-image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAeCAYAAAAVdY8wAAAABmJLR0QA/wD/AP+gvaeTAAAAmUlEQVQ4jWNgGM7g/2SGP/8nM/xBF2ci1oBRhcND4f9Ehsr/oQxChE10ZGhjMGF4RFjhcYatDNIM3P/jGKJwKWSEMf7PZvjH8JfhP8Nvhv8MDAwMjLkMLJgmMjAwMBxjyGTgZWBi4GVgxm01AwMD43yGmQzPGX4wsOJzIwwcZFDH7xmYqZsZHjFcZriJSzEG+F/AcIRoxbQHABATH4G/PExqAAAAAElFTkSuQmCC"
  height="30"
  width="10"
/>
`;
